verifier/src/report_body.rs

```diff
@@ -72,6 +77,41 @@ impl Verifier for AttributesVerifier<'_> {
     }
 }
 
+/// Verify the [`ConfigId`] is as expected.
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct ConfigIdVerifier<'a> {
+    expected_id: ConfigId,
+    report_body: &'a ReportBody,
+}
+
+impl<'a> ConfigIdVerifier<'a> {
+    /// Create a new [`ConfigIdVerifier`] instance.
+    ///
+    /// # Arguments:
+    /// * `expected_id` - The expected id.
+    /// * `report_body` - The report body containing config id that conforms
+    ///    to the `expected_id`.
+    pub fn new(expected_id: ConfigId, report_body: &'a ReportBody) -> Self {
+        Self {
+            expected_id,
+            report_body,
+        }
+    }
+}
+impl Verifier for ConfigIdVerifier<'_> {
+    type Error = VerificationError;
+    fn verify(&self) -> CtOption<Self::Error> {
+        let expected = self.expected_id.clone();
+        let actual = self.report_body.config_id();
+        // TODO - This should be a constant time comparison.
+        let is_some = if expected == actual { 0 } else { 1 };
```

Comment:
I think the next PR I'll work on getting rid of this TODO and the one for attributes. I wanted to have more than one item to feel out options.